examples/spinning_cube/src/lib.rs

```diff
@@ -1,68 +1,41 @@
 #[macro_use]
 extern crate gdnative as godot;
 
-use godot::init::{Property, PropertyHint, PropertyUsage};
-use godot::GodotString;
-
+#[derive(godot::NativeClass)]
+#[inherit(godot::MeshInstance)]
+#[user_data(godot::user_data::MutexData<RustTest>)]
+#[builder_helper{
+    builder.add_property(godot::init::Property {
+        name: "test/test_enum",
+        default: godot::GodotString::from_str("Hello"),
+        hint: godot::init::PropertyHint::Enum {
+            values: &["Hello", "World", "Testing"],
+        },
+        getter: |_: &RustTest| godot::GodotString::from_str("Hello"),
+        setter: (),
+        usage: godot::init::PropertyUsage::DEFAULT,
+    });
+    builder.add_property(godot::init::Property {
+        name: "test/test_flags",
+        default: 0,
+        hint: godot::init::PropertyHint::Flags {
+            values: &["A", "B", "C", "D"],
+        },
+        getter: |_: &RustTest| 0,
+        setter: (),
+        usage: godot::init::PropertyUsage::DEFAULT,
+    });
+}]
```

Comment:
This is not a good syntax, since it obfuscates the declaration with a large block up front, and is inherently unhygienic and hacky. Blocks within attributes are very unusual for the language, and the naming of this attribute is also less than satisfactory.

I suggest the following syntax that allows the user to name a custom register function:

```rust
// -snip -
#[register_with(my_register_function)]
struct Thing;

fn my_register_function(builder: &ClassBuilder<Thing>) {
    // - snip -
}
```

Reply:
That was a lot harder than expected but I finally got it down. The magic was to parse the tokens as `syn::BareFnArg`.

Reply:
That's strange. You shouldn't have a problem parsing it as a `syn::AttributeArgs`.

Reply:
Oh god thank you. You are 100% correct. I  seen the Attribute but not the AttributeArgs.